fv3core/fv3core/grid/generation.py

```diff
@@ -44,20 +44,10 @@
 from .mirror import mirror_grid
 
 
-# TODO remove this when using python 3.8+ everywhere, it comes for free
-def cached_property(func):
-    cached = None
-
-    @property
-    @functools.wraps(func)
-    def wrapped(*args, **kwargs):
-        nonlocal cached
-        if cached is None:
-            cached = func(*args, **kwargs)
-        return cached
-
-    return wrapped
-
+# TODO: when every environment in python3.8, replace
```

Comment:
we could even have:
```
def cached_property(f):
    @property
    @functools.lru_cache()
    def wrapper(self, *args, **kwargs):
        return f(self, *args, **kwargs)
    return wrapper
```
here and then the only change would be to get rid of it once we've moved up. Not super necessary

Reply:
I got really sick of trying out decorators and them not working right... I can do try this if you want